core/embed/rust/src/ui/component/model_t1/dialog.rs

```diff
@@ -0,0 +1,116 @@
+use super::{
+    button::{Button, ButtonMsg::Clicked},
+    theme,
+};
+use crate::{
+    micropython::buffer::Buffer,
+    ui::{
+        component::{Child, Component, Event, EventCtx},
+        display,
+        geometry::{Point, Rect},
+    },
+};
+
+pub enum DialogMsg<T> {
+    Content(T),
+    LeftClicked,
+    RightClicked,
+}
+
+pub struct Dialog<T, U> {
+    header: Option<Buffer>,
```

Comment:
I know using generics in this case is a bit unfortunate, but using `Buffer` here couples the UI to the MicroPython module. This might be OK, I don't know, but I've tried to avoid it. What do you think?

Reply:
Agreed that we shouldn't be tied to MicroPython unless needed. The `AsRef<[u8]>` solution is actually pretty nice (I just need to figure out some better var name than U). Fixed in c7b68d269d9641146d843a7d2035a1ffdf500edc.